storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -230,6 +235,55 @@ public Optional<String> serverConfigName(String configName) {
+
+    private static final int DEFAULT_EMPTY_NODE_ID = -1;
+
+    private static final String METADATA_LOG_DIR_PROP = "metadata.log.dir";
+
+    private static final String PROCESS_ROLES_PROP = "process.roles";
+
+    private static final String INTER_BROKER_PROTOCOL_VERSION_PROP = "inter.broker.protocol.version";
+
+    private static final String BROKER_ID_GENERATION_ENABLE_PROP = "broker.id.generation.enable";
+
+    private static final String ZOOKEEPER_CONNECT_PROP = "zookeeper.connect";
+
+    private static final String INTER_BROKER_PROTOCOL_VERSION_PROP_DEFAULT_VALUE = MetadataVersion.latest().version();
+    private static final String LOG_DIR_DOC = "The directory in which the log data is kept (supplemental for " + LOG_DIR_PROP + " property)";
+    private static final String METADATA_LOG_DIR_DOC = "This configuration determines where we put the metadata log for clusters in KRaft mode. " +
+        "If it is not set, the metadata log is placed in the first log directory from log.dirs.";
+    private static final String INTER_BROKER_PROTOCOL_VERSION_DOC = "Specify which version of the inter-broker protocol will be used.\n" +
+        " This is typically bumped after all brokers were upgraded to a new version.\n" +
+        " Example of some valid values are: 0.8.0, 0.8.1, 0.8.1.1, 0.8.2, 0.8.2.0, 0.8.2.1, 0.9.0.0, 0.9.0.1 Check MetadataVersion for the full list.";
+
+    private static final String LOG_DIRS_DOC = "A comma-separated list of the directories where the log data is stored. If not set, the value in " + LOG_DIRS_PROP + " is used.";
+
+    private static final String BROKER_ID_DOC = "The broker id for this server. If unset, a unique broker id will be generated." +
+        "To avoid conflicts between ZooKeeper generated broker id's and user configured broker id's, generated broker ids " +
+        "start from " + MAX_RESERVED_BROKER_ID_PROP + " + 1.";
+
+    private static final String NODE_ID_DOC = "The node ID associated with the roles this process is playing when `process.roles` is non-empty. " +
+        "This is required configuration when running in KRaft mode.";
+    private static final String PROCESS_ROLES_DOC = "The roles that this process plays: 'broker', 'controller', or 'broker,controller' if it is both. " +
+        "This configuration is only applicable for clusters in KRaft (Kafka Raft) mode (instead of ZooKeeper). Leave this config undefined or empty for ZooKeeper clusters.";
+
+    private static final String BROKER_ID_GENERATION_ENABLE_DOC = "Enable automatic broker id generation on the server. When enabled the value configured for $MaxReservedBrokerIdProp should be reviewed.";
+
+    private static final String MAX_RESERVED_BROKER_ID_DOC = "Max number that can be used for a broker.id";
+
+    private static final String ZK_CONNECT_DOC = "Specifies the ZooKeeper connection string in the form <code>hostname:port</code> where host and port are the " +
```

Comment:
I don't think this information should be in LogConfig.

Reply:
@fvaleri scala version of StorageTool executes this code block https://github.com/apache/kafka/blob/4981fa939d588645401619bfc3e321dc523d10e7/core/src/main/scala/kafka/server/KafkaConfig.scala#L2181C5-L2181C25 Hence I added similar in LogConfig. wdyt ? So property has to be defined first and then we can perform validations.